### ISSUE TITLE
the side-bar takes a whole buffer

### ISSUE THREAD

**Issue by zakariakebairia** [rougier/nano-sidebar#2]
when i use the `nano-sidebar-toggle`, it takes a whole buffer instead of just the "side"
is there a way so i can configure it on  a specific width?

![2021-08-31-1630385433_990x644](https://user-images.githubusercontent.com/3621146/131444163-8aeeb43c-40c4-4d60-9d8a-355864958b02.png)

**Comment by rougier** [comment]
Well, it's not supposed to do that! What is your system?

**Comment by subego** [comment]
Same question.
GNU Emacs 27.2 (build 1, x86_64-apple-darwin20.5.0, Carbon Version 164 AppKit 2022.5)
<img width="266" alt="CleanShot 2021-08-31 at 14 18 59@2x" src="https://user-images.githubusercontent.com/3454566/131452004-32ab2925-c135-4201-9241-844f347db4d4.png">

**Comment by zakariakebairia** [comment]
> Well, it's not supposed to do that! What is your system?

**OS**: Gnu/Linux -- Artix (Arch based distro)
**Emacs**: GNU Emacs 27.2 (build 1, x86_64-pc-linux-gnu, GTK+ Version 3.24.27, cairo version 1.17.4) of 2021-03-26

**Comment by rougier** [comment]
Thanks. Can you check if a frame is actually created (and cover the initial frame) or if there's an error during the creation?

**Comment by zakariakebairia** [comment]
no, actually , when i activate the nano-side-bar, the screen blink with the nano-side-bar frame (the frame appear and immediately disappear). 
these are the messages i get
![2021-08-31-1630401702_715x103](https://user-images.githubusercontent.com/3621146/131478543-7f01d24b-71ec-4cae-a2d0-c7eb7159d586.png)

After that, I can use ibuffer to locate the frame.
![2021-08-31-1630401777_840x418](https://user-images.githubusercontent.com/3621146/131478586-e268fe45-b957-427a-8552-534bce4c80c2.png)

**Comment by rougier** [comment]
Ok. Can you try to add a message [here](https://github.com/rougier/nano-sidebar/blob/master/nano-sidebar.el#L262) to check if the `make-frame-visible` function is called?

**Comment by zakariakebairia** [comment]
alright, i've added this message like that
```elisp
    (when (not visible)
;;;;;;;;;;;;;;;;;the message;;;;;;;;;;;;;;;;;;;
      (message "Hello there")
;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
      (make-frame-visible sidebar)
      (modify-frame-parameters sidebar `((top    . 0)
                                         (height . 1.0)
                                         (width  . ,sidebar-width)
                                         (left   . ,(* -1 width))))
      (select-frame-set-input-focus selected-frame))
```
These are the messages I got
```
t
nano-sidebar
Hello there
dark
```

**Comment by rougier** [comment]
Ok. Do you know if there are actually two frames or only one after toggling the sidebar?  For example, is one covering the other?

**Comment by zakariakebairia** [comment]
no, only one frame!
i toggled the sidebar from the **scratch** buffer 

`
(nano-sidebar-toggle)   ;; Using C-x C-e
`

![2021-08-31-1630414414_791x327](https://user-images.githubusercontent.com/3621146/131506136-83924944-cbd7-4f4e-9787-b8e6a480ef96.png)

**Comment by rougier** [comment]
Can you try:
```
(make-frame `((parent-frame . ,(window-frame))
              (undecorated .  nil)
              (user-position . nil)
              (user-size . nil)
              (background-color . "black")
              (height . 12)
              (width  . 72)))
```

**Comment by zakariakebairia** [comment]
where should i put that
i use it as a configuration, and a black box appeared at the top left of the buffer.
![2021-08-31-1630419395_1086x520](https://user-images.githubusercontent.com/3621146/131519415-e3e8b657-cf25-4c0d-9851-58adbb3d24ee.png)

**Comment by zakariakebairia** [comment]
**PS**: when i execute `(nano-sidebar-toggle)`, i loose my cursor, it seems that the focus goes to the new sidebar.
I can restore my cursor by clicking again.

**Comment by rougier** [comment]
The command was a test to chekc whether you can create a child frame and apparently you can. By the way, are you using a tiloing window manager or emacs in full screen (that won't play well with the sidebar)?

**Comment by zakariakebairia** [comment]
yes, i'm using dwm

**Comment by rougier** [comment]
That might be the reason. Can you try with a floating emacs window?

**Comment by zakariakebairia** [comment]
i tried with a floating mode in dwm... same problem 🤔

**Comment by rougier** [comment]
What about the minimal example I posted above? I suspect that the new frame might be identified as new window by your wm and put somewhere else. Since this new frame is a child frame, I'm not sure how dwm would handle it. Can you try with a non-tiling window manager (if possible, don't break your configuration for the test)?

**Comment by DivineDominion** [comment]
> Can you try:
> 
> ```
> (make-frame `((parent-frame . ,(window-frame))
>               (undecorated .  nil)
>               (user-position . nil)
>               (user-size . nil)
>               (background-color . "black")
>               (height . 12)
>               (width  . 72)))
> ```

@rougier I tried on macOS Monterey with Emacs 28 today, and this spawns a sub-window that is attached to the main one, much like macOS sheets used to maintain relative position.

![image](https://user-images.githubusercontent.com/59080/150193090-77ef439e-6916-44a5-b576-7daed7ac9f5e.png)

Thing is, the sidebar also is a floating window. Much like old Mac OS X GUIs, or GIMP:

![Screen Shot 2022-01-19 at 19 36 47](https://user-images.githubusercontent.com/59080/150193166-c32abb36-0aa9-4512-8628-7ced1534c8e6.png)

Is there anything else you do to customize the display of Emacs frames on your Mac?

**Comment by rougier** [comment]
@DivineDominion Sorry for late answer. I think this the expected behavior. In my case (OSX Monterrey EmacsMac 27.1), I removed the title bar.
